citesphere/src/main/java/edu/asu/diging/citesphere/web/user/SyncItemController.java

```diff
@@ -1,26 +1,77 @@
 package edu.asu.diging.citesphere.web.user;
 
+import java.util.List;
+import java.util.Optional;
+
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.security.core.Authentication;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
 
 import edu.asu.diging.citesphere.core.exceptions.CannotFindCitationException;
 import edu.asu.diging.citesphere.core.exceptions.GroupDoesNotExistException;
 import edu.asu.diging.citesphere.core.model.IUser;
+import edu.asu.diging.citesphere.core.model.bib.ICitationGroup;
+import edu.asu.diging.citesphere.core.model.bib.impl.CitationGroup;
+import edu.asu.diging.citesphere.core.repository.bib.CitationGroupRepository;
 import edu.asu.diging.citesphere.core.service.ICitationManager;
+import edu.asu.diging.citesphere.core.zotero.IZoteroConnector;
 
 @Controller
 public class SyncItemController {
     
     @Autowired
     private ICitationManager citationManager;
+    
+    @Autowired
+    private IZoteroConnector zoteroConnector;
+    
+    @Autowired
+    private CitationGroupRepository groupRepository;
 
     @RequestMapping(value="/auth/group/{zoteroGroupId}/items/{itemId}/sync")
     public String getItem(Authentication authentication, Model model, @PathVariable("zoteroGroupId") String zoteroGroupId, @PathVariable("itemId") String itemId) throws GroupDoesNotExistException, CannotFindCitationException {
         citationManager.updateCitationFromZotero((IUser)authentication.getPrincipal(), zoteroGroupId, itemId);
         return "redirect:/auth/group/{zoteroGroupId}/items/{itemId}";
     }
+    
+    @RequestMapping(value = {"/auth/group/{zoteroGroupId}/items/sync", "/auth/group/{zoteroGroupId}/collection/{collectionId}/items/sync"})
+    public String getCitationsList(Authentication authentication, Model model,
+            @PathVariable("zoteroGroupId") String zoteroGroupId,
+            @PathVariable(value="collectionId", required=false) String collectionId,
+            @RequestParam(required = false, value = "page") String page,
+            @RequestParam(defaultValue = "title", required = false, value = "sort") String sort,
+            @RequestParam(required = false, value = "columns") List<String> columns) throws GroupDoesNotExistException {
+        Optional<CitationGroup> groupOptional = groupRepository.findById(new Long(zoteroGroupId));
+        
+        String redirectUrl = "redirect:/auth/group/{zoteroGroupId}/items?page=" + page + "&sort=" + sort;
+        String columnString = null;
+        if(columns.size() > 0 && !columns.get(0).equals("[]")) {
```

Comment:
this should not be necessary. Spring should parse the list for you. have you tried using an array instead of a list?

Reply:
I used list because it is passed as list from the original api : [[version, publicationTitle]]. And when columns is empty it passes list with empty list as an element like this: [[]]

Reply:
spring should parse it for you correctly. have you tried using an array?

Reply:
Yes, When I try to pass array of String then it read it as [[Ljava.lang.String;@3eae7f02] in original api